scripts/extensions/auto-tagging-widget/src/auto-tagging.tsx

```diff
@@ -395,26 +395,91 @@ export function getAutoTaggingComponent(superdesk: ISuperdesk, label: string) {
 
                                 const entities = items.filter((tag) => isEntity(tag));
                                 const entitiesGrouped = entities.groupBy((tag) => tag?.group.value);
-                                const entitiesGroupedAndSorted = entitiesGrouped.sortBy(
-                                    (_, key) => key!.toString().toLocaleLowerCase(),
-                                    (a, b) => a.localeCompare(b),
-                                );
+
+                                const entitiesGroupedAndSortedByConfig = entitiesGrouped
+                                    .filter((_, key) => hasConfig(key, this.iMatricsFields.entities))
+                                    .sortBy((_, key) => this.iMatricsFields.entities[key].order,
+                                        (a, b) => a - b);
+
+                                const entitiesGroupedAndSortedNotInConfig = entitiesGrouped
+                                    .filter((_, key) => !hasConfig(key, this.iMatricsFields.entities))
+                                    .sortBy((_, key) => key!.toString().toLocaleLowerCase(),
+                                        (a, b) => a.localeCompare(b));
+
+                                const entitiesGroupedAndSorted = entitiesGroupedAndSortedByConfig
+                                    .concat(entitiesGroupedAndSortedNotInConfig);
 
                                 const others = items.filter((tag) => isEntity(tag) === false);
                                 const othersGrouped = others.groupBy((tag) => tag.group.value);
 
-                                const othersGroupedAndSortedByConfig = othersGrouped
-                                    .filter((_, key) => hasConfig(key, this.iMatricsFields))
-                                    .sortBy((_, key) => this.iMatricsFields[key].order,
```

Comment:
Could you explain what wasn't working with previous implementation?

It looks to me that you have added sorting specifically for entities, but I thought it would be enough to modify how `entitiesGroupedAndSorted` or `othersGroupedAndSorted` are put together. Why were changes needed below this line?

Reply:
issue is that group of entites are not sorted and included, because in old implementation we showed all groups and entities are added at bottom of page. Now we need to include sorting of entities group also, and to be included in config (NTB want to add entitie group at top of list

Reply:
![image](https://user-images.githubusercontent.com/6507688/118117333-c8a6e780-b3eb-11eb-9e37-df74a6761ede.png)

and this group should on top:
![image](https://user-images.githubusercontent.com/6507688/118117374-d5c3d680-b3eb-11eb-9060-d7d64d3894ec.png)

Reply:
What about the second question?

Reply:
we need to sort entities as group and items in entities group separately